### PR TITLE
#805(1) change function name

### PR DESCRIPTION
**Pull request by fergie-nz**
Snapshot tweaks

### CODE REVIEW

File: database/buildSchema/35_file.sql [hunk 1]
```diff
@@ -12,6 +12,7 @@ CREATE TABLE public.file (
     is_output_doc boolean DEFAULT FALSE NOT NULL,
     is_internal_reference_doc boolean DEFAULT FALSE NOT NULL,
     is_external_reference_doc boolean DEFAULT FALSE NOT NULL,
+    file_missing boolean DEFAULT FALSE NOT NULL,
```
Comment (CarlosNZ) [inline database/buildSchema/35_file.sql:15]: I'd call it `is_missing` so it's consistent with the others

**Review by CarlosNZ — COMMENTED** [review]
Looks good. I haven't tested it out yet, will do shortly.

Just a suggestion about the field name.

File: database/migration/migrateData.ts [hunk 2]
```diff
@@ -409,6 +409,8 @@ const migrateData = async () => {
       ADD COLUMN IF NOT EXISTS is_internal_reference_doc boolean DEFAULT FALSE NOT NULL;
       ALTER TABLE file
       ADD COLUMN IF NOT EXISTS is_external_reference_doc boolean DEFAULT FALSE NOT NULL;
+      ALTER TABLE file
+      ADD COLUMN IF NOT EXISTS is_missing boolean DEFAULT FALSE NOT NULL;
```
Comment (CarlosNZ) [inline database/migration/migrateData.ts:413]: Good idea to try and bundle this in with existing. However, this is the migration for versions less than `v0.3.1`, and we're past that now, so it needs to go in the latest batch. 

Just put the mutation down the bottom, straight after the last one.

File: src/components/snapshots/takeSnapshot.ts [hunk 3]
```diff
@@ -175,7 +175,7 @@ const copyFiles = async (
   filePaths.push(...fileRecords.map((fileRecord) => fileRecord.thumbnailPath))
   const baseFilePaths = options.resetFiles ? await getBaseFiles(FILES_FOLDER) : []
 
-  for (const filePath of [...filePaths, ...baseFilePaths]) {
+  for (const filePath of [...filePaths]) {
```
Comment (CarlosNZ) [inline src/components/snapshots/takeSnapshot.ts:178]: You'll need to remove this from `useSnapshot` too, and then remove the `getBasePaths` function and all references to it.

I'll want to test it again after that just to be sure, but I think it'll be okay

**Review by CarlosNZ — CHANGES_REQUESTED** [review]

File: database/migration/migrateData.ts (hunk 2 above)
Comment (fergie-nz) [reply database/migration/migrateData.ts:413]: I can see that the most recent conditional migration statement is for lessthan 0.4.5,

Should I make a new conditional statement now?

Comment (CarlosNZ) [reply database/migration/migrateData.ts]: No, we're still less than 0.4.5. We're on "pre-release" 0.4.5-13, which is considered a pre-cursor to "0.4.5."

File: database/migration/migrateData.ts [hunk 4]
```diff
@@ -400,7 +400,7 @@ const migrateData = async () => {
     `)
 
     console.log(
-      ' - Adding columns to file table: to_be_deleted, is_internal_reference_doc, is_external_reference_doc'
+      ' - Adding columns to file table: to_be_deleted, is_internal_reference_doc, is_external_reference_doc, is_missing'
```
Comment (CarlosNZ) [inline database/migration/migrateData.ts:403]: Cool, can you just revert this line now that you've moved it further back down?

File: src/components/snapshots/useSnapshot.ts [hunk 5]
```diff
@@ -242,7 +242,7 @@ const copyFiles = async (snapshotFolder: string, fileRecords: ObjectRecord[] = [
   const snapshotFilesFolder = `${snapshotFolder}/files`
   const baseFilePaths = await getBaseFiles(snapshotFilesFolder)
```
Comment (CarlosNZ) [inline src/components/snapshots/useSnapshot.ts:243]: This line now needs to go too.

**Review by CarlosNZ — COMMENTED** [review]
Cool, I think this is pretty good, just those couple of little things.

Don't merge it yet, we'll merge your second one into here when it's ready, then I'll give it a proper test all as one before we merge it all back to develop.

Cheers.

**Comment by nmadruga** [comment]
Hello @fergie-nz, I was wondering about this PRs... Would it be a good idea to combine those in a single PR?

Just hoping to get PRs before the release **0.4.6** otherwise (if more time is required) we can move this to the next release

**Comment by CarlosNZ** [comment]
> Hello @fergie-nz, I was wondering about this PRs... Would it be a good idea to combine those in a single PR?

He's already done that -- I'm just waiting to have another look when I have time.

**Comment by nmadruga** [comment]
> > Hello @fergie-nz, I was wondering about this PRs... Would it be a good idea to combine those in a single PR?
> 
> He's already done that -- I'm just waiting to have another look when I have time.

Sounds good - Which PR has all #805 combined?

Can this one be closed?

**Comment by CarlosNZ** [comment]
> Sounds good - Which PR has all #805 combined?

Not sure. @fergie-nz ?
 
> Can this one be closed?

I dunno -- I'll let Fergus tidy it up. I'll plan to have another look at this tomorrow.

**Pull request closed without merging**